### DISCUSSION TITLE
How to detect the hash value of the computed data

### DISCUSSION THREAD

**Discussion by qqqqqqQwQ** [lschoe/mpyc#89] [Q&A] [answered]
I would like to ask about the issue of data tampering during the calculation process, for example, when using a CSV file as input in Id3gini. My idea is to save the hash value of the data using blockchain before performing calculations, and then check whether the hash value of the data has changed after the calculations to ensure the security of the computation. Is there any function in the mpyc development package that can be used to detect the hash value of the computed data?

**Comment by lschoe** [comment, marked as answer]
I'm note sure about the scenario here. In the demo `id3gini.py` all parties start out with a copy of the same dataset stored in a `.csv` file. But that's only to keep the demo simple. A realistic scenario is when all parties use a private `.csv` file and secret-share these between each other using `mpc.input()` to create a horizontally partitioned dataset. See the `kmsurvival.py` demo for an example, which is still a demo as each party uses a fragment of a globally available dataset, like this
```python
df = df[mpc.pid::m]  # simple partition of dataset between m parties
```
To validate these private inputs against changes during the computation does not seem very relevant?

**Reply by qqqqqqQwQ** [reply]
Thank you for your answer!
